documentation/src/components/component.ts

```diff
@@ -14,15 +14,79 @@ import { LayoutElement } from './layout';
 import componentStyles from './markdown.css';
 import { AppRouter } from '../router';
 import { TabList } from '../../../packages/tab-list';
+import docs from '../../custom-elements.json';
 
-let ComponentApiDocs: Map<string, TemplateResult>;
 let ComponentDocs: Map<string, TemplateResult>;
 
 enum TabValue {
     Api = 'api',
     Examples = 'examples',
 }
 
+interface JsDocTagParsed {
+    tag: string;
+    name?: string;
+    attribute?: string;
+    type?: string;
+    optional?: boolean;
+    default?: string;
+    description?: string;
+}
+
+type ParsedTagArray = JsDocTagParsed[];
+
+function sortByName(a: JsDocTagParsed, b: JsDocTagParsed) {
+    if (!a.name || !b.name) {
+        return 0;
+    }
+    if (a.name > b.name) {
+        return 1;
+    }
+    if (b.name > a.name) {
+        return -1;
+    }
+    return 0;
+}
+
+function buildTable(
```

Comment:
This replaces the pre-processing of the API Documentation that had previously been done. This approach also saves us ~15KB of payload, which is nice.